AsyncDisplayKit/ASCellNode.mm

```diff
@@ -204,6 +244,26 @@ - (void)__setHighlightedFromUIKit:(BOOL)highlighted;
   }
 }
 
+- (NSIndexPath *)indexPath
+{
+  ASDisplayNodeAssertMainThread();
+
+  UIScrollView *scrollView = self.scrollView;
+
+  ASDisplayNode *owningNode = scrollView.asyncdisplaykit_node;
+  if (_scrollViewType.isTableView) {
+    return [(ASTableView *)owningNode indexPathForNode:self];
+  } else if (_scrollViewType.isTableNode) {
+    return [(ASTableNode *)owningNode indexPathForNode:self];
+  } else if (_scrollViewType.isCollectionView) {
+    return [(ASCollectionView *)owningNode indexPathForNode:self];
```

Comment:
`scrollView indexPathForNode:`

That said, I think we should actually remove support for ASCollectionView and ASTableView from here, and only work with the ASTableNode and ASCollectionNode. 

The rationale here is,
- The implementation is much simpler.
- The views return index paths in the view-space, which are pretty useless to developers.